### PR TITLE
Feat/available after

### PR DESCRIPTION
**Pull request by cnlangzi**
## Changes
-

## Fixes
- fixed issues on https://github.com/0chain/zboxcli/pull/131

## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/zboxcli/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

Associated PRs (Link as appropriate):
- blobber: https://github.com/0chain/blobber/pull/600
- system_test: https://github.com/0chain/system_test/pull/216
- zboxcli: https://github.com/0chain/zboxcli/pull/194
- gosdk: https://github.com/0chain/gosdk/pull/418

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | feat/available_after  |
| zboxcli  | feat/available_after  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | feat/available_after  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: README.md [hunk 1]
```diff
@@ -11,64 +11,67 @@ zbox is a command line interface (CLI) tool to understand the capabilities of 0C
...
-        - [Storage SC configurations](#storage-sc-configurations)
-        - [Stake pool info](#stake-pool-info)
-        - [Lock tokens into stake pool](#lock-tokens-into-stake-pool)
-        - [Unlock tokens from stake pool](#unlock-tokens-from-stake-pool)
-        - [Stake pools info of user](#stake-pools-info-of-user)
-        - [Write pool info](#write-pool-info)
-        - [Lock tokens into write pool](#lock-tokens-into-write-pool)
-        - [Unlock tokens from write pool](#unlock-tokens-from-write-pool)
+    - [Global Flags](#global-flags)
+- [Commands](#commands)
```
Comment (peterlimg) [inline README.md]: Is this indent an intended changes or a mistake

File: cmd/share.go [hunk 2]
```diff
@@ -63,8 +67,21 @@ var shareCmd = &cobra.Command{
 			fmt.Println("Share revoked for client " + refereeClientID)
 		} else {
 			expiration, _ := cmd.Flags().GetInt64("expiration-seconds")
+			availableAfter := time.Now()
+
+			availableAfterInput, _ := cmd.Flags().GetString("available-after")
```
Comment (peterlimg) [inline cmd/share.go]: Check the error in case the input is invalid

File: cmd/upload.go [hunk 3]
```diff
@@ -285,7 +285,7 @@ var streamCmd = &cobra.Command{
 	},
 }
 
-func startChunkedUpload(cmd *cobra.Command, allocationObj *sdk.Allocation, localPath, thumbnailPath, remotePath string, encrypt bool, chunkSize int, attrs fileref.Attributes, statusBar sdk.StatusCallback, isUpdate bool) error {
+func startChunkedUpload(cmd *cobra.Command, allocationObj *sdk.Allocation, localPath, thumbnailPath, remotePath string, encrypt bool, chunkSize int, attrs fileref.Attributes, statusBar sdk.StatusCallback, isUpdate, isRepair bool) error {
```
Comment (peterlimg) [inline cmd/upload.go]: The arguments length is too long, please put them into a struct to make it clean and easier to read

File: cmd/share.go [hunk 4]
```diff
@@ -63,8 +67,21 @@ var shareCmd = &cobra.Command{
...
+
+			if len(availableAfterInput) > 0 {
+				if aa, err := common.ParseTime(availableAfter, availableAfterInput); err != nil {
+					PrintError(err.Error())
+					os.Exit(1)
+				} else {
+					availableAfter = *aa
+				}
```
Comment (peterlimg) [inline cmd/share.go:86]: The `else` is unnecessary here, can do like this:

```go
aa, err := common.ParseTime(...)
if err != nil {
  ...
  os.Exit(1)
  return
}

availableAfter = *aa
```

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: cmd/share.go (hunk 4 above)
Comment (cnlangzi) [reply cmd/share.go:86]: fixed

File: cmd/upload.go (hunk 3 above)
Comment (cnlangzi) [reply cmd/upload.go]: fixed

File: cmd/share.go (hunk 2 above)
Comment (cnlangzi) [reply cmd/share.go]: fixed

File: README.md (hunk 1 above)
Comment (cnlangzi) [reply README.md]: it is automatically formatted by vscode.

Comment (cnlangzi) [reply README.md:15]: it seems be fine. just leave them as they are.

Screen Shot 2022-03-23 at 20 55 17

Comment (cnlangzi) [reply README.md]: fixed. it is a new feature in vscode : auto-TOC based on markdown

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | feat/available_after  |
| zboxcli  | feat/available_after  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | feat/available_after  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by peterlimg — APPROVED** [review]
lgtm

**Pull request merged**